### PR TITLE
#168781692 add and view comment

### PR DESCRIPTION
**Pull request by ngireric123**
#### What does this PR do?
- Add and view comments 
#### Description of Task to be completed?
- add the possibility for line manager and owner of the request to add a comment on the request
- add the possibility for line manager and owner of the request to view a comment on the request
#### How should this be manually tested?
- Browse to the repo directory
- navigate to Ft-comment-168781692 branch
- Run `npm install`
- run `sequelize db:migrate`
- run `sequelize db:seed:all`
- run `npm run dev`
#### Any background context you want to provide?
**To add a comment**
- In Postman Provide a login bearer token
- Fill the body with comment field
- send a `POST` request to `http://localhost:<:port>/api/v1/comments/:id`

**To view a comment**
- In Postman Provide a login bearer token
- send a `GET` request to `http://localhost:<:port>/api/v1/comments/:id`
#### What are the relevant pivotal tracker stories?
[#168781692](https://www.pivotaltracker.com/story/show/168781692)
#### Screenshots (if appropriate)
image

#### Questions:
N/A

### CODE REVIEW

**Review by alainmateso — APPROVED** [review]
Great job Eric. Kindly change the message that you return when posting a comment and make it comment-related. I think the user is posting a comment not updating a request. Other than that, well done.

File: src/routes/api/comments.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+import commentMiddleware from '../../middlewares/commentMiddleware';
+
+const router = new Router();
+
+const { addComment, getComments } = commentsController;
+
+const { validateComment } = InputValidation;
+
+const { checkExistingRequest, checkOwner } = commentMiddleware;
+// const { checkExistingRequest } = commentMiddleware;
```
Comment (alainmateso) [inline src/routes/api/comments.js]: Remove this commented line

**Review by sabin18 — APPROVED** [review]

File: src/controllers/commentsController.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import models from '../database/models';
+import responseUtil from '../utils/responseUtil';
+import strings from '../utils/stringsUtil';
+
+export default class CommentsController {
+
+  static async addComment(req, res) {
+    const { comment } = req.body;
+    const { id } = req.user.payload;
+    const requestId = parseInt(req.params.id, 10);
```
Comment (nakiwuge) [inline src/controllers/commentsController.js:12]: Add a new line here

File: src/controllers/commentsController.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        ...data
+      } = addedComment.dataValues;
+
+      return responseUtil(res, 200, strings.request.success.SUCCESS_ADD_COMMENT, data);
+    } catch (error) {
+      return res.send(error);
+    }
+  }
+
+
```
Comment (nakiwuge) [inline src/controllers/commentsController.js]: Remove this line

File: src/controllers/commentsController.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  static async getComments(req, res) {
+    try {
+      const requestId = parseInt(req.params.id, 10);
+      const comments = await models.comments.findAll({
+        where: { requestId }
```
Comment (nakiwuge) [inline src/controllers/commentsController.js]: This gets all comments on a request even the soft deleted ones. Add a condition to get comments only when the `deleted= false`

File: src/middlewares/commentMiddleware.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-irregular-whitespace */
+import responseError from '../utils/responseError';
+import string from '../utils/stringsUtil';
+import models from '../database/models';
+
+const { requests, users } = models;
+
+const commentMiddle = {
```
Comment (nakiwuge) [inline src/middlewares/commentMiddleware.js]: Get this a better name please

File: src/routes/api/comments.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { Router } from 'express';
+import commentsController from '../../controllers/commentsController';
+import validateToken from '../../middlewares/auth/validateToken';
+import InputValidation from '../../middlewares/inputValidation';
+import commentMiddleware from '../../middlewares/commentMiddleware';
+
+const router = new Router();
+
```
Comment (nakiwuge) [inline src/routes/api/comments.js]: remove this line

File: src/routes/api/comments.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+const { addComment, getComments } = commentsController;
+
```
Comment (nakiwuge) [inline src/routes/api/comments.js]: remove this line

File: src/routes/api/comments.js [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+const { validateComment } = InputValidation;
+
```
Comment (nakiwuge) [inline src/routes/api/comments.js:11]: remove this line

File: src/tests/comment.spec.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+const token = generateToken(mockdata.verifiedUser1);
+const token2 = generateToken(mockdata.requester);
+const token3 = generateToken(mockdata.verifiedUser);
+
+chai.should();
+chai.use(chaiHttp);
+
+const id = mockdata.verifiedUser1.id;
+
```
Comment (nakiwuge) [inline src/tests/comment.spec.js:16]: remove this line

File: src/controllers/commentsController.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        deleted, createdAt, updatedAt, userId,
+        ...data
+      } = addedComment.dataValues;
+
+      return responseUtil(res, 200, strings.request.success.SUCCESS_ADD_COMMENT, data);
+    } catch (error) {
+      return res.send(error);
+    }
+  }
+
```
Comment (nakiwuge) [inline src/controllers/commentsController.js:27]: remove this line

**Review by nakiwuge — CHANGES_REQUESTED** [review]
@ngireric123 good job. Kindly work on the feedback below.
- Return the user object you can return `id, username and email` when retrieving comments
- Add your endpoints to swagger
- Check out the comments I have left on the pr

**Review by nakiwuge — APPROVED** [review]

**Pull request merged**